inc/commonitilobject.class.php

```diff
@@ -6921,9 +6921,17 @@ function showTimeline($rand) {
 
                echo "<span class='h_user_name'>";
                $userdata = getUserName($item_i['users_id'], 2);
-               echo $user->getLink()."&nbsp;";
-               echo Html::showToolTip($userdata["comment"],
-                                      ['link' => $userdata['link']]);
+               if (\Entity::getUsedConfig('anonymize_support_agents')
+                  && \Session::getCurrentInterface() == 'helpdesk'
+                  && $item_i['timeline_position'] == self::TIMELINE_RIGHT) {
```

Comment:
I'm not entirely sure for this condition; relying on timeline position sounds wrong

Reply:
The only downside I can think of is if a tech forget to assign himself and then add a followup -> not anonymized because timeline position == 1 (but i guess this is his fault then ?).

Do we have a better way to check this ?

Reply:
No I do not have a way for now, I'll check.

To me, it is quite complicated to understand that `$item_i['timeline_position'] == self::TIMELINE_RIGHT` means "technician" :D

Reply:
Done in e849947.

Reply:
Plus you added unit tests... :+1: